### PR TITLE
[CARBONDATA-3188] Create carbon table as hive understandable metastore table needed by Presto and Hive

### PR DESCRIPTION
**Pull request by ravipesala**
Problem:
Current carbon table created in spark creates the hive table internally but it does not have much information like schema, input/output format and location details. So other execution engines like Presto and Hive cannot read the table.

Reason:
Spark always checks in HiveSerde static map whether it is a hive supported table or not, since carbon is not registered to that map it cannot create hive understandable table. It justs creates a table without schema and location and adds its own schema as part of properties.

Solution:
Add the carbon details also to HiveSerde static map so that it can create Hive understandable table.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1854/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1855/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1856/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10111/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2064/

**Comment by SteNicholas** [comment]
@ravipesala This pull request doesn't work in #3012 .HiveExample run exception through your changes.But this update does work in beeline.Therefore will you check the cause of HiveExample exception with me？

**Comment by ravipesala** [comment]
@SteNicholas This PR does not work directly on the Hive integration as we need to set the right serde, inputformat and output format. We are planning to refactor the current Hive integration code to unify the inpuformat and outputformat with the hadoop modules input/outformats.  This PR is basic step for both Hive and Presto integration.

**Comment by SteNicholas** [comment]
@ravipesala Could I refactor the current Hive integration code with you?

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  override def afterAll(): Unit = {
+    sql("DROP TABLE IF EXISTS source")
+  }
+
+  test("test create table and verify the hive table correctness with stored by") {
+    sql("DROP TABLE IF EXISTS source")
+    sql(
+      s"""
+         |CREATE TABLE source (key INT, value string, col1 double)
+         |STORED BY 'carbondata'
```
Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala]: please use STORED AS carbondata

Comment (jackylk) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala]:
```suggestion
         |STORED AS carbondata
```

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1961/

**Comment by SteNicholas** [comment]
@jackylk OK.I'd like to refactory on carbondata-hive module based on @ravipesala update.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    sql("DROP TABLE IF EXISTS source")
+  }
+
+  private def verifyTable = {
+    val table = sqlContext.sparkSession.asInstanceOf[CarbonSession].sessionState.catalog.asInstanceOf[CarbonSessionCatalog].getClient().getTable("default", "source")
+    assert(table.schema.fields.length == 3)
+    if (SparkUtil.isSparkVersionEqualTo("2.2")) {
+      assert(table.storage.locationUri.get.equals(new Path(s"file:$storeLocation/source").toUri))
+    }
+    assert(table.storage.inputFormat.get.equals(classOf[CarbonFileInputFormat[_]].getName))
```
[the remark this answers is not in the document]
Comment (ravipesala) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala]: OK, changed to CarbonTableInputFormat.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  test("test create table and verify the hive table correctness with using carbon") {
+    sql("DROP TABLE IF EXISTS source")
+    sql(
+      s"""
+         |CREATE TABLE source (key INT, value string, col1 double)
+         |using carbon
+     """.stripMargin)
+
+    verifyTable
+
```
[the remark this answers is not in the document]
Comment (ravipesala) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala]: ok

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    sql("DROP TABLE IF EXISTS source")
+    sql(
+      s"""
+         |CREATE TABLE source (key INT, value string, col1 double)
+         |using carbondata
+     """.stripMargin)
+
+    verifyTable
+
+
```
[the remark this answers is not in the document]
Comment (ravipesala) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala]: ok

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+     """.stripMargin)
+
+    verifyTable
+
+    sql("DROP TABLE IF EXISTS source")
+  }
+
+  private def verifyTable = {
+    val table = sqlContext.sparkSession.asInstanceOf[CarbonSession].sessionState.catalog.asInstanceOf[CarbonSessionCatalog].getClient().getTable("default", "source")
+    assert(table.schema.fields.length == 3)
```
[the remark this answers is not in the document]
Comment (ravipesala) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala]: ok

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala (hunk 1 above)
Comment (ravipesala) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateHiveTableWithCarbonDS.scala]: ok

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10213/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2234/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**